### PR TITLE
chore: inherit PBS & Slurm defaults 

### PR DESCRIPTION
**Pull request by phillip-gaisford**
## Description

Inherit PBS & Slurm defaults from task container defaults for launcher-provided pools.

- Fixes an existing problem where PBS & Slurm arguments were not being acted upon.
- Brings in task container defaults from launcher-provided pools.
- New RM-level API for resolving task container defaults.

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by jerryharrow — COMMENTED** [review]
lgtm, but should get Bradley's approval.

File: master/internal/api_command.go [hunk 1]
```diff
@@ -103,17 +103,18 @@ func (a *apiServer) getCommandLaunchParams(ctx context.Context, req *protoComman
 		return nil, launchWarnings, fmt.Errorf("checking respurce availability: %v", err.Error())
 	}
 	// Get the base TaskSpec.
-	taskContainerDefaults := a.m.getTaskContainerDefaults(poolName)
+	taskContainerDefaults := a.m.rm.TaskContainerDefaults(
+		a.m.system,
+		resources.ResourcePool,
+		a.m.config.TaskContainerDefaults,
```
Comment (stoksc) [inline master/internal/api_command.go:109]: maybe ideally this wouldn't get passed, but i guess i need to think about the abstractions some more

**Review by stoksc — APPROVED** [review]
lgtm

File: master/internal/rm/agentrm/agent_resource_manager.go [hunk 2]
```diff
@@ -735,3 +754,19 @@ func (a ResourceManager) NotifyContainerRunning(
...
+
+// TaskContainerDefaults returns TaskContainerDefaults for the specified pool.
+func (a ResourceManager) TaskContainerDefaults(
+	ctx actor.Messenger,
+	pool string,
+	defaultConfig model.TaskContainerDefaultsConfig,
+) model.TaskContainerDefaultsConfig {
+	request := taskContainerDefaults{fallbackDefault: defaultConfig, resourcePool: pool}
+	response := a.Ref().System().Ask(a.Ref(), request).Get()
+	return response.(model.TaskContainerDefaultsConfig)
```
Comment (stoksc) [inline master/internal/rm/agentrm/agent_resource_manager.go:771]: oh oops, this should use a.Ask

**Review by stoksc — CHANGES_REQUESTED** [review]

File: master/internal/rm/agentrm/agent_resource_manager.go (hunk 2 above)
Comment (stoksc) [reply master/internal/rm/agentrm/agent_resource_manager.go:771]: same in the dispatcher ee code. if you peruse the code for https://github.com/determined-ai/determined/blob/master/master/internal/rm/actorrm/actor_resource_manager.go#L265-L291 it should sort of explain why (what it handles that this doesn't)

File: master/internal/rm/agentrm/agent_resource_manager.go [hunk 3]
```diff
@@ -735,3 +754,20 @@ func (a ResourceManager) NotifyContainerRunning(
...
+func (a ResourceManager) TaskContainerDefaults(
+	ctx actor.Messenger,
+	pool string,
+	fallbackConfig model.TaskContainerDefaultsConfig,
+) (model.TaskContainerDefaultsConfig, error) {
+	result := model.TaskContainerDefaultsConfig{}
+	err := a.Ask(ctx,
+		taskContainerDefaults{fallbackDefault: fallbackConfig, resourcePool: pool}, &result)
+	return result, err
+}
```
Comment (stoksc) [inline master/internal/rm/agentrm/agent_resource_manager.go:773]:
```suggestion
) (result model.TaskContainerDefaultsConfig, err error) {
        req := taskContainerDefaults{fallbackDefault: fallbackConfig, resourcePool: pool}
	return result, a.Ask(ctx, req, &result)
}
```

**Review by stoksc — APPROVED** [review]
lgtm

**Pull request merged**